### PR TITLE
WIP: Feature/collection navigation

### PR DESCRIPTION
**Pull request by 3rfan**

### CODE REVIEW

File: husky_components/collection-navigation/collections-list-view.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+define(['text!./collections-list.html'], function(collectionsListTpl) {
```
Comment (wachterjohannes) [inline husky_components/collection-navigation/collections-list-view.js]: add the header here!

File: husky_components/collection-navigation/main.js [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    },
+
+    namespace = 'husky.collection-navigation.',
+
+    /**
+     * Creates the eventnames
+     * @param postfix {String} event name to append
+     */
+    createEventName = function(postfix) {
+        return namespace + postfix;
```
Comment (wachterjohannes) [inline husky_components/collection-navigation/main.js]: shouldn´t we use an instance name here?

File: husky_extensions/cache-factory.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
+(function() {
+
+    'use strict';
+
+    define([], {
+        name: 'history',
```
Comment (3rfan) [inline husky_extensions/cache-factory.js]: rename to cache-factory

File: husky_components/collection-navigation/main.js [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+
+    var defaultOptions = {
+        url: null,
+        collectionId: null,
+        translates: {
+            noCollections: ''
+        }
+    },
+
+    consts = {
```
Comment (wachterjohannes) [inline husky_components/collection-navigation/main.js]: dont use abbreviations

File: husky_components/collection-navigation/main.js [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+        /**
+         * @method parse
+         * @param {Object} response Response
+         */
+        parse: function(response) {
+            var current = {},
+                parent = response._embedded.parent;
+
+            current.id = response.id || consts.ROOT_ID;
+            current.name = response.name;
```
Comment (wachterjohannes) [inline husky_components/collection-navigation/main.js]: in the root there is no name please provide a default value which is translated (can be set over options).

Additionally change name to a configurable property name (default title)

File: husky_components/collection-navigation/main.js [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+
+            if (!!parent) {
+                parent.id = parent.id || consts.ROOT_ID;
+            }
+
+            this.data = {
+                children: response._embedded.items || null,
```
Comment (wachterjohannes) [inline husky_components/collection-navigation/main.js]: change `items` to a configurable property name (default items)

File: husky_components/collection-navigation/main.js [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+         * @method getCollectionItems
+         */
+        getCollectionItems: function(id) {
+            var dfd = $.Deferred(),
+                data = this.cache.get(id),
+                item, url;
+
+            if (!data) {
+                if (id === consts.ROOT_ID) {
+                    url = this.data.parent._links.self.href;
```
Comment (wachterjohannes) [inline husky_components/collection-navigation/main.js]: change self to a configurable property name (default children)

File: husky_components/collection-navigation/main.js [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+                } else {
+                    // underscores where returns a list but we only want the first item 
+                    item = _.where(this.data.children, { id: id })[0];
+                    url = item._links.self.href;
```
Comment (wachterjohannes) [inline husky_components/collection-navigation/main.js]: configurable property name

File: husky_components/collection-navigation/collections-list.html [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        <div class="collection-navigation-loader-container">
+            <div class="spinner">
+                <div class="double-bounce1"></div>
+                <div class="double-bounce2"></div>
+            </div>
+        </div>
+    </li>
+    <% } %>
+</ul>
+<div class="collection-navigation-list-footer">
```
Comment (wachterjohannes) [inline husky_components/collection-navigation/collections-list.html]: please provide a flag to display or hide this button

File: husky_extensions/cache-factory.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
+(function() {
```
Comment (wachterjohannes) [inline husky_extensions/cache-factory.js]: header (=

**Comment by wachterjohannes** [comment]
@3rfan i have discussed with @berndhep this changes should be done:

* click on image/name go into it and load childs
* click on arrow ony load childs
* flag hasSub indicates that the collection has no child or has (=
* click on a navigation witzhout child should show an empty list this adds the abbility to add a child to it

File: husky_components/collection-navigation/main.js (hunk 2 above)
Comment (wachterjohannes) [inline husky_components/collection-navigation/main.js]: yes we should use (= please added it example from overlay ....

**Comment by wachterjohannes** [comment]
the collection-navigation is now named data-navigation and is build into the normal navigation as a "third-layer".

the open issues now (after discuss with bernd):

__Move the cross into the navigation__

from: 
image

to:
image

Replace cross with arrow left.

__set `navigation-data-container` width to 300px__

__move data-navigation over content if navigation is uncollapsing__

**Comment by 3rfan** [comment]
Maybe we should change the tooltip color? @berndhep 
image

**Comment by berndhep** [comment]
@3rfan Make it in the color of the main navigation background.

File: apimocker_config.json [hunk 11]
```diff
@@ -4,7 +4,7 @@
     "quiet": false,
     "port": "7878",
     "latency": 700,
-    "allowedDomains": ["http://husky.lo"],
+    "allowedDomains": ["*"],
```
Comment (danrot) [inline apimocker_config.json]: Best change ever.

File: husky_components/data-navigation/list-view.js [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+                '<li class="not-selectable">',
+                '<div class="data-navigation-loader-container">',
+                '<div class="spinner">',
+                '<div class="double-bounce1"></div>',
+                '<div class="double-bounce2"></div>',
+                '</div>',
+                '</div>',
+                '</li>',
+                '<% } %>',
+                '</ul>'
```
Comment (danrot) [inline husky_components/data-navigation/list-view.js]: Wrong indentation.

And yeah, @3rfan I know you would like to have that in separate files :-P

File: husky_components/data-navigation/main.js [hunk 13]
```diff
@@ -0,0 +1,536 @@
...
+                    '<% } else { %>',
+                    '<%= translates.title %>',
+                    '<% } %>',
+                    '</div>',
+                    '</div>',
+                    '<% } else { %>',
+                    '<div class="root-text">',
+                    '<%= data.current.name || translates.title %>',
+                    '</div>',
+                    '<% } %>'
```
Comment (danrot) [inline husky_components/data-navigation/main.js]: Wrong indentation

File: husky_components/data-navigation/main.js [hunk 14]
```diff
@@ -0,0 +1,536 @@
...
+                    '<div class="data-navigation-list-container"></div>',
+                    '<% if (options.showAddButton) { %>',
+                    '<div class="data-navigation-list-footer">',
+                    '<button class="data-navigation-add btn">',
+                    '<span class="fa-plus-circle"></span>',
+                    '<%= options.translates.addButton %>',
+                    '</button>',
+                    '</div>',
+                    '<% } %>',
+                    '</div>'
```
Comment (danrot) [inline husky_components/data-navigation/main.js]: wrong indentation

File: husky_components/data-navigation/main.js [hunk 15]
```diff
@@ -0,0 +1,536 @@
...
+            return createEventName.call(this, 'clear-cache')
+        };
+
+    return {
+        /**
+         * @method initialize
+         */
+        initialize: function() {
+            this.currentView = null;
+            this.cache = this.sandbox.cacheFactory.create();
```
Comment (danrot) [inline husky_components/data-navigation/main.js]: Just curios: what exactly are you caching here?

File: husky_components/navigation/main.js [hunk 16]
```diff
@@ -539,6 +558,33 @@ define(function() {
...
+                parent;
+
+            if (this.sandbox.dom.hasClass(domObject, 'is-selected')) {
+                return;
+            }
+
+            if (this.sandbox.dom.hasClass(domObject, 'js-navigation-sub-item')) {
+                parent = this.sandbox.dom.closest(domObject, '.navigation-items');
+
+                // toggle parent only when it is not expaneded
```
Comment (danrot) [inline husky_components/navigation/main.js]: s/expaneded/expanded

**Comment by danrot** [comment]
Could this be merged without https://github.com/sulu-cmf/sulu/pull/812, and everything would continue to work as expected? /cc @wachterjohannes

**Comment by wachterjohannes** [comment]
@danrot i would do this the same like the sulu pr freeze it and merge it with the next changes!

File: apimocker_config.json (hunk 11 above)
Comment (wachterjohannes) [inline apimocker_config.json]: haha (-: it has bothered me for a long time

File: husky_components/data-navigation/main.js (hunk 15 above)
Comment (wachterjohannes) [inline husky_components/data-navigation/main.js]: we cache the navigation responses that you can go thru the layer faster ...

File: husky_components/data-navigation/list-view.js (hunk 12 above)
Comment (wachterjohannes) [inline husky_components/data-navigation/list-view.js]: (= fixed

**Comment by chirimoya** [comment]
@3rfan please make PR mergeable

**Pull request merged**